flink-metrics/flink-metrics-prometheus/src/test/java/org/apache/flink/metrics/prometheus/PrometheusReporterTest.java

```diff
@@ -192,32 +150,31 @@ void histogramIsReportedAsPrometheusSummary() throws UnirestException {
         }
     }
 
+    /**
+     * Metrics with the same name are stored by the reporter in a shared data-structure. This test
+     * ensures that a metric is unregistered from Prometheus even if other metrics with the same
+     * name still exist.
+     */
     @Test
-    void metricIsRemovedWhenCollectorIsNotUnregisteredYet() throws UnirestException {
-        JobManagerMetricGroup jmMetricGroup =
-                JobManagerMetricGroup.createJobManagerMetricGroup(registry, HOST_NAME);
-
+    void metricIsRemovedWhileOtherMetricsWithSameNameExist() throws UnirestException {
         String metricName = "metric";
 
         Counter metric1 = new SimpleCounter();
-        FrontMetricGroup<JobManagerJobMetricGroup> metricGroup1 =
-                new FrontMetricGroup<>(
-                        createReporterScopedSettings(),
-                        jmMetricGroup.addJob(JobID.generate(), "job_1"));
-        reporter.notifyOfAddedMetric(metric1, metricName, metricGroup1);
-
         Counter metric2 = new SimpleCounter();
-        FrontMetricGroup<JobManagerJobMetricGroup> metricGroup2 =
-                new FrontMetricGroup<>(
-                        createReporterScopedSettings(),
-                        jmMetricGroup.addJob(JobID.generate(), "job_2"));
-        reporter.notifyOfAddedMetric(metric2, metricName, metricGroup2);
 
-        reporter.notifyOfRemovedMetric(metric1, metricName, metricGroup1);
+        final Map<String, String> variables2 = new HashMap<>(metricGroup.getAllVariables());
+        final Map.Entry<String, String> entryToModify = variables2.entrySet().iterator().next();
+        final String labelValueThatShouldBeRemoved = entryToModify.getValue();
+        variables2.put(entryToModify.getKey(), "some_value");
+        final MetricGroup metricGroup2 = TestUtils.createTestMetricGroup(LOGICAL_SCOPE, variables2);
+
+        reporter.notifyOfAddedMetric(metric1, metricName, metricGroup);
+        reporter.notifyOfAddedMetric(metric2, metricName, metricGroup2);
+        reporter.notifyOfRemovedMetric(metric1, metricName, metricGroup);
 
         String response = pollMetrics(reporter.getPort()).getBody();
 
-        assertThat(response).doesNotContain("job_1");
+        assertThat(response).doesNotContain(labelValueThatShouldBeRemoved);
```

Comment:
Why is it enough here to just check for the removal? Shouldn't we also check for the other metric not being touched? 🤔